### DISCUSSION TITLE
How to set active color to submenu header

### DISCUSSION THREAD

**Discussion by for-dev9** [tailwindlabs/tailwindcss#14757] [Help]
Is it posible to set active color to 'Parent' sub menu ?

<li><a>Item 1</a></li>
<li>
    <details open>
      <summary class='active'>**Parent**</summary>
      <ul>
        <li><a>Submenu 1</a></li>
        <li><a>Submenu 2</a></li>
      </ul>
    </details>
  </li>

**Comment by wongjn** [comment]
Yes, I would assume so. Consider providing a [Tailwind Play](https://play.tailwindcss.com) with your code and give more details if you'd like more concrete help.

**Comment by aniruddhaadak80** [comment]
Hi @for-dev9 ,

Yes, it's possible to set an active color to the 'Parent' submenu header. You can achieve this by applying a specific class to the parent item when it's active. Here's an example using CSS:

```css
.menu-item.active {
    color: #007bff; /* Change to your desired active color */
}
```

Then, in your HTML, add the `active` class to the parent item when it's selected:

```html
<ul class="menu">
    <li class="menu-item active">Parent</li>
    <li class="menu-item">Submenu 1</li>
    <li class="menu-item">Submenu 2</li>
</ul>
```

You can adjust the styling and logic to fit your specific framework or setup. If you have a JavaScript framework, you can toggle the `active` class based on the current route or user interaction.

Thanks!

**Comment by UtkarshK95** [comment]
Yes, you can set an active color to the "Parent" submenu header by using a conditional class based on whether the submenu is active. Here are a couple of approaches depending on your setup:

### 1. **Using Tailwind and JavaScript (React Example)**

If you're using a framework like React, you can conditionally apply a class to the "Parent" based on the active state of the submenu.

```jsx
import { useState } from 'react';

const Menu = () => {
  const [isSubmenuActive, setIsSubmenuActive] = useState(false);

  return (
    <div>
      <div className="mb-2">Item 1</div>
      <div 
        onClick={() => setIsSubmenuActive(!isSubmenuActive)} 
        className={`cursor-pointer ${isSubmenuActive ? 'text-blue-500' : 'text-gray-800'}`}
      >
        Parent
      </div>
      {isSubmenuActive && (
        <div className="ml-4">
          <div>Submenu 1</div>
          <div>Submenu 2</div>
        </div>
      )}
    </div>
  );
};
```

In this example:
- The "Parent" header toggles its text color between blue and gray when clicked.
- `text-blue-500` is applied when the submenu is active, while `text-gray-800` is the default.

### 2. **Using CSS Only**

If you're using only HTML and CSS, you can apply an active color to the submenu header with CSS by toggling a class when the submenu is open:

```html
<style>
  .submenu-header {
    color: gray;
    cursor: pointer;
  }
  .submenu-header.active {
    color: blue;
  }
</style>

<div>
  <div>Item 1</div>
  <div class="submenu-header" onclick="toggleSubmenu(this)">
    Parent
  </div>
  <div class="submenu" style="display: none;">
    <div>Submenu 1</div>
    <div>Submenu 2</div>
  </div>
</div>

<script>
  function toggleSubmenu(header) {
    const submenu = header.nextElementSibling;
    submenu.style.display = submenu.style.display === 'none' ? 'block' : 'none';
    header.classList.toggle('active');
  }
</script>
```

Here:
- JavaScript toggles the `active` class on the "Parent" header when the submenu is shown/hidden.
- The `.active` class changes the color to blue while `.submenu-header` sets the default gray color. 

This way, the active color changes based on the visibility of the submenu without needing a framework like React.
